### PR TITLE
HZC-2715: Serverless cluster support

### PR DESCRIPTION
**Pull request by sdreger**
New features:
- Support for creating serverless clusters

### CODE REVIEW

File: models/cluster.go [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
+	// Small cluster type.
+	Small ClusterType = "SMALL"
+	// Medium cluster type.
+	Medium ClusterType = "MEDIUM"
+	// Large cluster type.
+	Large ClusterType = "LARGE"
+	// Serverless cluster type.
+	Serverless ClusterType = "SERVERLESS"
+	// Devmode cluster type.
+	Devmode ClusterType = "DEVMODE"
```
Comment (pivovarit) [inline models/cluster.go:18]: I think DEVMODE is a subclass of SERVERLESS and not really a separate cluster type - to be confirmed tomorrow

File: serverless_cluster.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+		Response:  cluster,
+	}
+	req, err := svc.client.NewRequest(&graphqlRequest)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	resp, err := svc.client.Do(ctx, req, &cluster)
+	if err != nil {
+		return nil, resp, err
```
Comment (pivovarit) [inline serverless_cluster.go]: `resp` is `nil` here I think (just like a few lines above)

File: testdata/access_token_response.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"data":{"response":{"token":"hidden"}}}
```
Comment (pivovarit) [inline testdata/access_token_response.json:1]: Wouldn't it be easier to just pass it as a string?

Comment (sdreger) [reply testdata/access_token_response.json:1]: I just wanted to be consistent and store all the mocked response files in the `testdata` folder.
But if you think it's better to use a string here - that's also fine.

File: models/cluster.go (hunk 1 above)
Comment (sdreger) [reply models/cluster.go:18]: In our current implementation we have a separate cluster type: `DEV_MODE`

**Comment by huseyinbabal** [comment]
@sdreger thanks for your implementation, I tested it and it works, please see my note in your jira ticket here https://hazelcast.atlassian.net/browse/HZC-2715
You can ping me once again once you make this PR ready
Well done 🎖️

**Review by huseyinbabal — APPROVED** [review]

File: serverless_cluster.go [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+	if err != nil {
+		return nil, nil, err
+	}
+
+	resp, err := svc.client.Do(ctx, req, &clusterList)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return &clusterList, resp, err
```
Comment (ivan-shevtsov) [inline serverless_cluster.go:63]: does it make sense to create a separate task to do postfiltering by plan_name? and also to make request in graphql unpaged.

**Review by ivan-shevtsov — APPROVED** [review]

File: serverless_cluster.go (hunk 4 above)
Comment (sdreger) [reply serverless_cluster.go:63]: I think this is the question to @huseyinbabal . As long we are going to disable `standard` clusters, this method should return both `standard` and `serverless` clusters. But in fact will return just `serverless`. But anyway I'd like to know what @huseyinbabal thinks about that.

**Pull request merged**